### PR TITLE
Timeout parameter for wait and rename rm to delete

### PR DESCRIPTION
**Pull request by dhodovsk**

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/bWDs](https://da.gd/bWDs)__
* Comment: __See details for logs__

File: conu/apidefs/container.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def create(cls, image, container_params):
 
     def start(self):
         """
-        start current container
+        start current container - the container has to be running
```
Comment (TomasTomecek) [inline conu/apidefs/container.py]: has to be created, not running (you shouldn't work so late :D)

File: conu/apidefs/container.py [hunk 2]
```diff
@@ -273,10 +273,11 @@ def get_status(self):
         """
         raise NotImplementedError("get_status is not implemented")
 
-    def wait(self):
+    def wait(self, timeout):
         """
         Block until the container stops, then return its exit code.
 
+        :param timeout: Request timeout
```
Comment (TomasTomecek) [inline conu/apidefs/container.py]: please write down the expected type (int/float)

File: conu/backend/docker/container.py [hunk 3]
```diff
@@ -267,7 +267,7 @@ def run_via_binary_in_foreground(
 
     def start(self):
         """
-        start current container
+        start current container - the container has to be running
```
Comment (TomasTomecek) [inline conu/backend/docker/container.py]: the same

File: conu/backend/docker/container.py [hunk 4]
```diff
@@ -341,11 +341,12 @@ def get_status(self):
         """
         return self.get_metadata(refresh=True)["State"]["Status"]
 
-    def wait(self):
+    def wait(self, timeout=None):
         """
         Block until the container stops, then return its exit code. Similar to
         the ``docker wait`` command.
 
+        :param timeout: Request timeout
```
Comment (TomasTomecek) [inline conu/backend/docker/container.py]: the same

File: conu/apidefs/container.py (hunk 1 above)
Comment (dhodovsk) [reply conu/apidefs/container.py]: Oh my, you're right! :D

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/q5PL](https://da.gd/q5PL)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/W3vOn](https://da.gd/W3vOn)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/BcIGg](https://da.gd/BcIGg)__
* Comment: __See details for logs__

**Pull request merged**